src/blueprint/azext_blueprint/custom.py

```diff
@@ -480,8 +481,9 @@ def list_blueprint_version_artifact(cmd, client, blueprint_name,
 
 
 def _del_artifact_name(rg):
-    del rg['artifact_name']
-    return rg
+    rg_copy = rg.copy()
+    del rg_copy['artifact_name']
+    return rg_copy

```

Comment:
What's the story about this change?

Reply:
https://github.com/Azure/azure-cli-extensions/blob/30be073a9f3b954a7724fa841e218195c24cfb8b/src/blueprint/azext_blueprint/custom.py#L528

In Python 3.6 and 3.7 `_del_artifact_name(rg)` is executed before `rg['artifact_name']` while Python 3.8 has the expected order, so a key error is thrown in 3.6 and 3.7.

Reply:
It was a bug in Python 3.6 and 3.7 to evaluate values first in a dictionary comprehension and fixed in 3.8: https://stackoverflow.com/a/42203486/2100487

Reply:
Really tricky. I don't think it's a good practice to rely on the evaluation order of key and value. Maybe you can split it into multiple lines. `rg.copy()` has a extra overhead.

Reply:
Updated.